File: leetcode/1513.cpp
class Solution {
  public:
    int numSub(string s) {
        auto contiguous_ones = s | views::split('0');

        function<long(int)> triangular =
            [](int order) {
                long a = long(order);
                return a * (a + 1) / 2;
            };

        return ranges::fold_left(
            contiguous_ones,
            0,
            [triangular] (int count, auto ones_range) {
                int ones = string_view(ones_range).size();

                return (count + triangular(ones)) % 1000000007;
            }
        );
    }
};

File: leetcode/1513.py
class Solution:
    def numSub(self, s: str) -> int:
        contiguous_ones = ( len(ones) for ones in s.split('0') if ones )
        triangular = lambda a: a * (a + 1) // 2
        
        return reduce(
            lambda count, ones: (count + triangular(ones)) % 1000000007,
            contiguous_ones,
            0
        )
